external/llvm-project/mlir/lib/Dialect/Tosa/Transforms/TosaPartition.cpp

```diff
@@ -374,10 +377,22 @@ void outlineConvPartOps(Operation *convOp, ArrayRef<Operation *> secondOps,
   }
 }
 
+bool isConv2D(Operation *op) { return isa<tosa::Conv2DOp>(op); }
+
 // Inspired by / adapted from TestSCFIfUtilsPass in
 // test/lib/Transforms/TestSCFUtils.cpp.
 class TosaPartitionPass : public TosaPartitionBase<TosaPartitionPass> {
+  std::function<bool(Operation *)> isPrimaryOp;
+
 public:
+  TosaPartitionPass() = default;
+  TosaPartitionPass(std::function<bool(Operation *)> pred,
+                    const std::string &attrName, bool nofront)
+      : isPrimaryOp(std::move(pred)) {
+    attributeName = attrName;
```

Comment:
Something about type conversions required that I assign these here rather than initialise them like isPrimaryOp.

Reply:
Can we go for `llvm::function_ref<>` instead of `std::function<>`?

Reply:
I did not know that was a thing.  I made the change.